crates/proto/src/rr/rdata/caa.rs

```diff
@@ -134,15 +140,47 @@ impl CAA {
         self.tag = tag;
     }
 
-    /// a potentially associated value with the property tag, see struct documentation
+    /// A value associated with the property tag, see struct documentation
+    #[deprecated = "See value_as_issue(), value_as_iodef(), or raw_value() instead"]
     pub fn value(&self) -> &Value {
         &self.value
     }
 
     /// Set the value associated with the property tag, see struct documentation
     pub fn set_value(&mut self, value: Value) {
+        self.raw_value = match &value {
+            Value::Issuer(name, key_values) => encode_issuer_value(name.as_ref(), key_values),
+            Value::Url(url) => url.as_str().as_bytes().to_vec(),
+            Value::Unknown(value) => value.clone(),
+        };
         self.value = value;
     }
+
+    /// Get the value of an `issue` or `issuewild` CAA record.
+    ///
+    /// This returns an error if the record's tag is not `issue` or `issuewild`, or if the value
+    /// does not match the expected syntax.
+    pub fn value_as_issue(&self) -> ProtoResult<(Option<Name>, Vec<KeyValue>)> {
```

Comment:
So I think the intent here is that we can express these in terms of `raw_value` in the future, right, and this is more like an optimization where we've already parsed a `Value` because of backwards compatibility concerns, right? Perhaps we should add a comment about this? (Same for `value_as_iodef()`?)